File: lobster-input-ios/LobsterKeyboard/Typing/Composing/ComposingTextBridge.swift
import UIKit

/// 将中文拼音组合串同步到目标输入框的下划线 composing 区域(setMarkedText)。
/// 与键盘麦克风 ASR 预览互斥:录音中暂停拼音 composing。
final class ComposingTextBridge {

    private var proxyProvider: (() -> UITextDocumentProxy?)?
    private var micActiveProvider: (() -> Bool)?
    private var lastPreview = ""

    func bind(
        proxyProvider: @escaping () -> UITextDocumentProxy?,
        micActiveProvider: @escaping () -> Bool
    ) {
        self.proxyProvider = proxyProvider
        self.micActiveProvider = micActiveProvider
    }

    // 搜狗式:拼音只在键盘候选区显示,不写入目标输入框(否则拼音字母会插进用户输入框)。
    func update(composingDisplay: String) { /* no-op */ }

    func clear() { /* no-op */ }
}

File: lobster-input-android/app/src/main/java/com/lobster/input/keyboard/typing/composing/ComposingTextBridge.kt
package com.lobster.input.keyboard.typing.composing

import android.view.inputmethod.InputConnection

/**
 * 拼音组合串的目标框同步(桥)。
 *
 * 搜狗式设计:**拼音只在键盘候选区显示,绝不写入目标输入框**——
 * 否则打字时拼音字母会插进用户输入框,选词又追加(如「237449测试」)。
 * 键盘麦克风的实时 ASR 预览走独立的 setComposingText 路径(不经过本桥)。
 */
class ComposingTextBridge {

    fun bind(
        connectionProvider: () -> InputConnection?,
        micActiveProvider: () -> Boolean
    ) { /* 不再需要:拼音不进目标框 */ }

    fun update(composingDisplay: String) { /* no-op:拼音只在候选区显示 */ }

    fun clear() { /* no-op */ }
}
